pkg/security/oauth2/plgd.go

```diff
@@ -12,14 +12,21 @@ import (
 
 // NewPlgdProvider creates OAuth client
 func NewPlgdProvider(ctx context.Context, config Config, logger log.Logger) (*PlgdProvider, error) {
-	config.ResponseMode = "query"
-	config.AccessType = "offline"
-	config.ResponseType = "code"
+	if config.ResponseMode == "" {
+		config.ResponseMode = "query"
+	}
+	if config.AccessType == "" {
+		config.AccessType = "offline"
+	}
+	if config.ResponseType == "" {
+		config.ResponseType = "code"
+	}
 
 	clientSecret, err := file.Load(config.ClientSecretFile, make([]byte, 4096))
 	if err != nil {
 		return nil, err
 	}
+	config.ClientSecret = string(clientSecret)
```

Comment:
@jkralik could you check this PR please, without this the CTT Origin tests don't work. 

Seems OK to me, but fields `ClientSecret`, `AuthURL` and `TokenURL` seems are part of config, but also part of oauth2 (line 42) and now they have the same value. Maybe there's a way to do it better and also make the code in `addLinkedAccount.go` work, but I do not know how.

Reply:
@Danielius1922 
lines 29 and 40,41,42 are not needed to be modified/added if you apply the comment in addLinkedAccount.go. So could you change it and test if it works?

Reply:
We can add call AuthCodeURL to provider config. Instead of `rh.provider.Config.AuthCodeURL(t)` we can call `rh.provider.AuthCodeURL(t)`

Reply:
I'm going to get rid of the whole `OAuth2` substruct as we talked about, it'll make things cleaner.